core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1692,43 +1697,60 @@ write_files:
           apiGroup: rbac.authorization.k8s.io
 {{ end }}
 
-  - path: /etc/kubernetes/manifests/kube-proxy.yaml
+  - path: /srv/kubernetes/manifests/kube-proxy-ds.yaml
     content: |
-        apiVersion: v1
-        kind: Pod
+        apiVersion: extensions/v1beta1
+        kind: DaemonSet
         metadata:
           name: kube-proxy
           namespace: kube-system
           labels:
             k8s-app: kube-proxy
           annotations:
             rkt.alpha.kubernetes.io/stage1-name-override: coreos.com/rkt/stage1-fly
-
         spec:
-          hostNetwork: true
-          containers:
-          - name: kube-proxy
-            image: {{.HyperkubeImage.RepoWithTag}}
-            command:
-            - /hyperkube
-            - proxy
-            - --master=http://127.0.0.1:8080
-            securityContext:
-              privileged: true
-            volumeMounts:
-            - mountPath: /etc/ssl/certs
-              name: ssl-certs-host
-              readOnly: true
-            - mountPath: /var/run/dbus
-              name: dbus
-              readOnly: false
-          volumes:
-          - hostPath:
-              path: /usr/share/ca-certificates
-            name: ssl-certs-host
-          - hostPath:
-              path: /var/run/dbus
-            name: dbus
+          updateStrategy:
+            type: RollingUpdate
+          template:
+            metadata:
+              labels:
+                k8s-app: kube-proxy
+              annotations:
+                scheduler.alpha.kubernetes.io/critical-pod: ''
+            spec:
+              serviceAccountName: kube-proxy
+              tolerations:
+              - operator: Exists
+                effect: NoSchedule
+              - operator: Exists
+                effect: NoExecute
+              - operator: Exists
+                key: CriticalAddonsOnly
+              hostNetwork: true
+              containers:
+              - name: kube-proxy
+                image: {{.HyperkubeImage.RepoWithTag}}
+                command:
+                - /hyperkube
+                - proxy
+                - --kubeconfig=/etc/kubernetes/proxy-kubeconfig.yaml
```

Comment:
can we avoid mounting /etc/kubernetes into pod? There are ssl certs which this pod really dont need